lib/auth/auth_with_roles_test.go

```diff
@@ -5519,3 +5519,165 @@ func TestSafeToSkipInventoryCheck(t *testing.T) {
+				mustResourceID(srv.ClusterName(), types.KindRole, requestRole.GetName()),
+			}),
+			errAssertionFunc: require.NoError,
+			expected: mustAccessRequest(t, alice, types.RequestState_PENDING, []string{requestRole.GetName()}, []types.ResourceID{
+				mustResourceID(srv.ClusterName(), types.KindRole, requestRole.GetName()),
+			}),
+		},
+		{
+			name: "bob fails to create a request for alice",
+			user: bob,
+			accessRequest: mustAccessRequest(t, alice, types.RequestState_PENDING, []string{requestRole.GetName()}, []types.ResourceID{
+				mustResourceID(srv.ClusterName(), types.KindRole, requestRole.GetName()),
+			}),
+			errAssertionFunc: require.Error,
+		},
+		{
+			name: "user creates own pending access request with user group needing app expansion",
+			user: alice,
+			accessRequest: mustAccessRequest(t, alice, types.RequestState_PENDING, []string{requestRole.GetName()}, []types.ResourceID{
+				mustResourceID(srv.ClusterName(), types.KindRole, requestRole.GetName()),
+				mustResourceID(srv.ClusterName(), types.KindUserGroup, userGroup1.GetName()),
+				mustResourceID(srv.ClusterName(), types.KindApp, "app1"),
+				mustResourceID(srv.ClusterName(), types.KindUserGroup, userGroup2.GetName()),
+				mustResourceID(srv.ClusterName(), types.KindUserGroup, userGroup3.GetName()),
+			}),
+			errAssertionFunc: require.NoError,
+			expected: mustAccessRequest(t, alice, types.RequestState_PENDING, []string{requestRole.GetName()}, []types.ResourceID{
+				mustResourceID(srv.ClusterName(), types.KindRole, requestRole.GetName()),
+				mustResourceID(srv.ClusterName(), types.KindUserGroup, userGroup1.GetName()),
+				mustResourceID(srv.ClusterName(), types.KindApp, "app1"),
+				mustResourceID(srv.ClusterName(), types.KindUserGroup, userGroup2.GetName()),
+				mustResourceID(srv.ClusterName(), types.KindUserGroup, userGroup3.GetName()),
+				mustResourceID(srv.ClusterName(), types.KindApp, "app2"),
+				mustResourceID(srv.ClusterName(), types.KindApp, "app3"),
+				mustResourceID(srv.ClusterName(), types.KindApp, "app4"),
+				mustResourceID(srv.ClusterName(), types.KindApp, "app5"),
+			}),
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			// Make sure there are no access requests before we do anything
+			ctx := context.Background()
+			require.NoError(t, srv.Auth().DeleteAllAccessRequests(ctx))
+
+			client, err := srv.NewClient(TestUser(test.user))
+			require.NoError(t, err)
+
+			test.errAssertionFunc(t, client.CreateAccessRequest(ctx, test.accessRequest))
+
+			accessRequests, err := srv.Auth().GetAccessRequests(ctx, types.AccessRequestFilter{
+				ID: test.accessRequest.GetName(),
+			})
+			require.NoError(t, err)
+
+			if test.expected == nil {
+				require.Empty(t, accessRequests)
+				return
+			}
+
+			require.Len(t, accessRequests, 1)
+
+			require.Empty(t, cmp.Diff(test.expected, accessRequests[0],
+				cmpopts.IgnoreFields(types.Metadata{}, "Name", "ID", "Expires"),
```

Comment:
Can you comment on why these fields are ignored? ID sounds irrelevant but Expires sounds like something important.

Reply:
I've managed to add in most of these fields for equivalence, so only Name and ID are ignored now. I've added a comment as to why name is ignored.